core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -1384,7 +1386,9 @@ public Builder setPreviousFileLocation(String previousFileLocation) {
     private boolean hasChanges() {
       return changes.size() != startingChangeCount
           || (discardChanges && !changes.isEmpty())
-          || metadataLocation != null;
+          || metadataLocation != null
+          || suppressHistoricalSnapshots
+          || null != snapshotsSupplier;
```

Comment:
this is for the client-side so that it doesn't exit early in https://github.com/apache/iceberg/blob/f19643a93f5dac99bbdbc9881ef19c89d7bcd3eb/core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java#L359-L368